src/AbstractSQLCompiler.ts

```diff
@@ -349,21 +553,25 @@ const compileSchema = (
 	ruleStatements = _.map(
 		abstractSqlModel.rules,
 		(rule): SqlRule => {
-			let ruleBody = _.find(rule, { 0: 'Body' } as Partial<AbstractSqlQuery>);
-			if (ruleBody == null || _.isString(ruleBody)) {
+			const ruleBodyNode = _.find(rule, { 0: 'Body' }) as [
+				'Body',
+				AbstractSqlQuery
+			];
```

Comment:
Is it worth exporting this as `RuleBodyNode`?

Reply:
I avoided exporting these two as they are a special case that only works for `abstractSqlModel.rules` and so we don't want them being typed as valid for the more common query case - they need to get covered in another pass where we eventually get rid of the generic `AbstractSqlQuery` type altogether